### PR TITLE
update test-infra to generate bool fields in ci-op-reference

### PR DESCRIPTION
**Pull request by smg247**
Specifically to get https://github.com/kubernetes/test-infra/commit/81c1f517b04d6467717008f9816703263a5794e2.

```
go get k8s.io/test-infra
make update-vendor
```
I then had to make a change to pass in an empty slice in `cmd/payload-testing-prow-plugin/server.go#trustedUsers` as the `trigger.TrustedUser` parameters changed.

merge after:
https://github.com/openshift/ci-chat-bot/pull/245
https://github.com/openshift/release-controller/pull/370

### CODE REVIEW

**Comment by smg247** [comment]
/label tide/merge-method-squash

...this has been an epic journey

File: pkg/webreg/zz_generated.ci_operator_reference.go [hunk 1]
```diff
@@ -29,6 +29,8 @@ const ciOperatorReferenceYaml = "# The list of base images describe\n" +
 	"# the pipeline will caches on. The one way is to take the reference\n" +
 	"# from an image stream, and the other from a dockerfile.\n" +
 	"build_root:\n" +
+	"    # If the BuildRoot images pullspec should be read from a file in the repository (BuildRootImageFileName).\n" +
```
Comment (hongkailiu) [inline pkg/webreg/zz_generated.ci_operator_reference.go:32]: I have no idea where it comes from except `make generate` updates it.

**Review by hongkailiu — APPROVED** [review]

File: pkg/webreg/zz_generated.ci_operator_reference.go (hunk 1 above)
Comment (hongkailiu) [reply pkg/webreg/zz_generated.ci_operator_reference.go:32]: I finally understand it.
This is what you did on the upstream.
Nice.

**Pull request merged**